### ISSUE TITLE
aws athena get-query-results  NextToken is not provided

### ISSUE THREAD

**Issue by jguerrero77** [aws/aws-cli#2756]
Hi there!
I just updated from aws cli 1.11.89 to 1.11.131

It turns out that the athena command get-query-results is not giving back any NextToken. According to the documentation we should be expecting it, but nothing is provided.
Any hints?

Thanks in advance.

**Comment by joguSD** [comment]
Can you post debug logs for this request? You can add the `--debug` flag to turn them on.

**Comment by jguerrero77** [comment]
Thanks for you quick response. Please, find the log at the end of this message. Although NextToken appears multiple times, the output does not include the NextToken field...

bash-4.1$ aws athena get-query-results --query-execution-id "c0b503b0-efa7-426e-aaa7-e1a4d7bcaa5e" --debug >> outputdata.txt
2017-08-11 08:11:54,714 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.11.131 Python/2.6.6 Linux/2.6.32-504.8.1.el6.x86_64 botocore/1.5.94
2017-08-11 08:11:54,714 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['athena', 'get-query-results', '--query-execution-id', 'c0b503b0-efa7-426e-aaa7-e1a4d7bcaa5e', '--debug']
2017-08-11 08:11:54,714 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_scalar_parsers at 0x1d9eed8>
2017-08-11 08:11:54,714 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x1a3aed8>
2017-08-11 08:11:54,715 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/research51/.local/lib/python2.6/site-packages/botocore/data/athena/2017-05-18/service-2.json
2017-08-11 08:11:54,733 - MainThread - botocore.hooks - DEBUG - Event service-data-loaded.athena: calling handler <function register_retries_for_service at 0x1531488>
2017-08-11 08:11:54,734 - MainThread - botocore.handlers - DEBUG - Registering retry handlers for service: athena
2017-08-11 08:11:54,735 - MainThread - botocore.hooks - DEBUG - Event building-command-table.athena: calling handler <function add_waiters at 0x1db6c80>
2017-08-11 08:11:54,739 - MainThread - awscli.clidriver - DEBUG - OrderedDict([(u'query-execution-id', <awscli.arguments.CLIArgument object at 0x214ee10>), (u'next-token', <awscli.arguments.CLIArgument object at 0x214ee50>), (u'max-results', <awscli.arguments.CLIArgument object at 0x214ee90>)])
2017-08-11 08:11:54,739 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.athena.get-query-results: calling handler <function add_streaming_output_arg at 0x1da3d70>
2017-08-11 08:11:54,739 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.athena.get-query-results: calling handler <function add_cli_input_json at 0x1a42410>
2017-08-11 08:11:54,740 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.athena.get-query-results: calling handler <function unify_paging_params at 0x1d24b18>
2017-08-11 08:11:54,743 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/research51/.local/lib/python2.6/site-packages/botocore/data/athena/2017-05-18/paginators-1.json
2017-08-11 08:11:54,744 - MainThread - awscli.customizations.paginate - DEBUG - Modifying paging parameters for operation: GetQueryResults
2017-08-11 08:11:54,744 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.athena.get-query-results: calling handler <function add_generate_skeleton at 0x1cca848>
2017-08-11 08:11:54,744 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.athena.get-query-results: calling handler <bound method CliInputJSONArgument.override_required_args of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x214ef50>>
2017-08-11 08:11:54,744 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.athena.get-query-results: calling handler <bound method GenerateCliSkeletonArgument.override_required_args of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x21513d0>>
2017-08-11 08:11:54,745 - MainThread - botocore.hooks - DEBUG - Event operation-args-parsed.athena.get-query-results: calling handler <functools.partial object at 0x2033f70>
2017-08-11 08:11:54,745 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.query-execution-id: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,746 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.athena.get-query-results: calling handler <awscli.argprocess.ParamShorthandParser object at 0x1db7590>
2017-08-11 08:11:54,746 - MainThread - awscli.arguments - DEBUG - Unpacked value of u'c0b503b0-efa7-426e-aaa7-e1a4d7bcaa5e' for parameter "query_execution_id": u'c0b503b0-efa7-426e-aaa7-e1a4d7bcaa5e'
2017-08-11 08:11:54,746 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.next-token: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,746 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.max-results: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,746 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.cli-input-json: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,746 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.starting-token: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,746 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.page-size: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,747 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.max-items: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,747 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.athena.get-query-results.generate-cli-skeleton: calling handler <function uri_param at 0x19756e0>
2017-08-11 08:11:54,747 - MainThread - botocore.hooks - DEBUG - Event calling-command.athena.get-query-results: calling handler <bound method CliInputJSONArgument.add_to_call_parameters of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x214ef50>>
2017-08-11 08:11:54,747 - MainThread - botocore.hooks - DEBUG - Event calling-command.athena.get-query-results: calling handler <bound method GenerateCliSkeletonArgument.generate_json_skeleton of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x21513d0>>
2017-08-11 08:11:54,747 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: env
2017-08-11 08:11:54,747 - MainThread - botocore.credentials - INFO - Found credentials in environment variables.
2017-08-11 08:11:54,748 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/research51/.local/lib/python2.6/site-packages/botocore/data/endpoints.json
2017-08-11 08:11:54,792 - MainThread - botocore.client - DEBUG - Registering retry handlers for service: athena
2017-08-11 08:11:54,793 - MainThread - botocore.hooks - DEBUG - Event creating-client-class.athena: calling handler <function add_generate_presigned_url at 0x14d2758>
2017-08-11 08:11:54,793 - MainThread - botocore.args - DEBUG - The s3 config key is not a dictionary type, ignoring its value of: None
2017-08-11 08:11:54,797 - MainThread - botocore.endpoint - DEBUG - Setting athena timeout as (60, 60)
2017-08-11 08:11:54,798 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.athena.GetQueryResults: calling handler <function generate_idempotent_uuid at 0x152fe60>
2017-08-11 08:11:54,799 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=GetQueryResults) (verify_ssl=True) with params: {'body': '{"QueryExecutionId": "c0b503b0-efa7-426e-aaa7-e1a4d7bcaa5e"}', 'url': u'https://athena.us-east-1.amazonaws.com/', 'headers': {'User-Agent': 'aws-cli/1.11.131 Python/2.6.6 Linux/2.6.32-504.8.1.el6.x86_64 botocore/1.5.94', 'Content-Type': u'application/x-amz-json-1.1', 'X-Amz-Target': u'AmazonAthena.GetQueryResults'}, 'context': {'auth_type': None, 'client_region': 'us-east-1', 'has_streaming_input': False, 'client_config': <botocore.config.Config object at 0x223c550>}, 'query_string': '', 'url_path': '/', 'method': u'POST'}
2017-08-11 08:11:54,799 - MainThread - botocore.hooks - DEBUG - Event request-created.athena.GetQueryResults: calling handler <bound method RequestSigner.handler of <botocore.signers.RequestSigner object at 0x223c450>>
2017-08-11 08:11:54,799 - MainThread - botocore.hooks - DEBUG - Event choose-signer.athena.GetQueryResults: calling handler <function set_operation_specific_signer at 0x152f578>
2017-08-11 08:11:54,800 - MainThread - botocore.auth - DEBUG - Calculating signature using v4 auth.
2017-08-11 08:11:54,800 - MainThread - botocore.auth - DEBUG - CanonicalRequest:
POST
/

content-type:application/x-amz-json-1.1
host:athena.us-east-1.amazonaws.com
x-amz-date:20170811T081154Z
x-amz-target:AmazonAthena.GetQueryResults

content-type;host;x-amz-date;x-amz-target
fa296a3b09fccc0842f4b50817aa1bbdc3c2d2f311761c1915432320a971ea05
2017-08-11 08:11:54,800 - MainThread - botocore.auth - DEBUG - StringToSign:
AWS4-HMAC-SHA256
20170811T081154Z
20170811/us-east-1/athena/aws4_request
796aae5a19fe238ea8327baa626dbbde011c395b179ef03d8742eb71386c4de5
2017-08-11 08:11:54,800 - MainThread - botocore.auth - DEBUG - Signature:
695127782964ce26b4fe38690dc58a8b877864224bc17937923562cda23ed04a
2017-08-11 08:11:54,806 - MainThread - botocore.endpoint - DEBUG - Sending http request: <PreparedRequest [POST]>
2017-08-11 08:11:54,806 - MainThread - botocore.vendored.requests.packages.urllib3.connectionpool - INFO - Starting new HTTPS connection (1): athena.us-east-1.amazonaws.com
2017-08-11 08:11:55,970 - MainThread - botocore.vendored.requests.packages.urllib3.connectionpool - DEBUG - "POST / HTTP/1.1" 200 47587
2017-08-11 08:11:56,087 - MainThread - botocore.parsers - DEBUG - Response headers: {'date': 'Fri, 11 Aug 2017 08:11:55 GMT', 'x-amzn-requestid': 'bd238a03-7e6c-11e7-b673-cb5194eedde5', 'content-length': '47587', 'content-type': 'application/x-amz-json-1.1', 'connection': 'keep-alive'}
2017-08-11 08:11:56,088 - MainThread - botocore.parsers - DEBUG - Response body:
{"NextToken":"AQKMs8AZPARBN3pfVLuY3RE7u3Y8bFO8zzSfSsrhcQekCrA+EdUnZ10SjoOUB1j5pJTNYhY0Sjy2c1N4XYb3Zj3Zs0LDJKj4bQ==","ResultSet":{"ColumnInfos":[{"CaseSensitive":false,"CatalogName":"hive","Label":"_col0","Name":"_col0","Nullable":"UNKNOWN","Precision":10,"Scale":0,"SchemaName":"","TableName":"","Type":"integer"}],"ResultRows":[{"Data":["_col0"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data"... <truncated>
2017-08-11 08:11:56,190 - MainThread - botocore.hooks - DEBUG - Event needs-retry.athena.GetQueryResults: calling handler <botocore.retryhandler.RetryHandler object at 0x214e050>
2017-08-11 08:11:56,190 - MainThread - botocore.retryhandler - DEBUG - No retry needed.
2017-08-11 08:11:56,191 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.athena.GetQueryResults: calling handler <function generate_idempotent_uuid at 0x152fe60>
2017-08-11 08:11:56,191 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=GetQueryResults) (verify_ssl=True) with params: {'body': '{"NextToken": "AQKMs8AZPARBN3pfVLuY3RE7u3Y8bFO8zzSfSsrhcQekCrA+EdUnZ10SjoOUB1j5pJTNYhY0Sjy2c1N4XYb3Zj3Zs0LDJKj4bQ==", "QueryExecutionId": "c0b503b0-efa7-426e-aaa7-e1a4d7bcaa5e"}', 'url': u'https://athena.us-east-1.amazonaws.com/', 'headers': {'User-Agent': 'aws-cli/1.11.131 Python/2.6.6 Linux/2.6.32-504.8.1.el6.x86_64 botocore/1.5.94', 'Content-Type': u'application/x-amz-json-1.1', 'X-Amz-Target': u'AmazonAthena.GetQueryResults'}, 'context': {'auth_type': None, 'client_region': 'us-east-1', 'has_streaming_input': False, 'client_config': <botocore.config.Config object at 0x223c550>}, 'query_string': '', 'url_path': '/', 'method': u'POST'}
2017-08-11 08:11:56,191 - MainThread - botocore.hooks - DEBUG - Event request-created.athena.GetQueryResults: calling handler <bound method RequestSigner.handler of <botocore.signers.RequestSigner object at 0x223c450>>
2017-08-11 08:11:56,191 - MainThread - botocore.hooks - DEBUG - Event choose-signer.athena.GetQueryResults: calling handler <function set_operation_specific_signer at 0x152f578>
2017-08-11 08:11:56,191 - MainThread - botocore.auth - DEBUG - Calculating signature using v4 auth.
2017-08-11 08:11:56,192 - MainThread - botocore.auth - DEBUG - CanonicalRequest:
POST
/

content-type:application/x-amz-json-1.1
host:athena.us-east-1.amazonaws.com
x-amz-date:20170811T081156Z
x-amz-target:AmazonAthena.GetQueryResults

content-type;host;x-amz-date;x-amz-target
cda784f4f7f697d78fc3aaba752f6302412a61ac62c89f8f7c0e0512f55a5208
2017-08-11 08:11:56,192 - MainThread - botocore.auth - DEBUG - StringToSign:
AWS4-HMAC-SHA256
20170811T081156Z
20170811/us-east-1/athena/aws4_request
54194d8274ea0d58921dd0dcd61e565b6ba66f1dd0c1b1ce882f3d54e668ee85
2017-08-11 08:11:56,192 - MainThread - botocore.auth - DEBUG - Signature:
2ae8eb76a61dfa9b7c3982f73d9cc7a358f260272536e7ac4ef4e9f5ebed38e3
2017-08-11 08:11:56,192 - MainThread - botocore.endpoint - DEBUG - Sending http request: <PreparedRequest [POST]>
2017-08-11 08:11:56,468 - MainThread - botocore.vendored.requests.packages.urllib3.connectionpool - DEBUG - "POST / HTTP/1.1" 200 981
2017-08-11 08:11:56,469 - MainThread - botocore.parsers - DEBUG - Response headers: {'date': 'Fri, 11 Aug 2017 08:11:55 GMT', 'x-amzn-requestid': 'bdb6064a-7e6c-11e7-bc3b-f5a2c5735062', 'content-length': '981', 'content-type': 'application/x-amz-json-1.1', 'connection': 'keep-alive'}
2017-08-11 08:11:56,469 - MainThread - botocore.parsers - DEBUG - Response body:
{"ResultSet":{"ColumnInfos":[{"CaseSensitive":false,"CatalogName":"hive","Label":"_col0","Name":"_col0","Nullable":"UNKNOWN","Precision":10,"Scale":0,"SchemaName":"","TableName":"","Type":"integer"}],"ResultRows":[{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]},{"Data":["1"]}],"ResultSetMetadata":{"ColumnInfo":[{"CaseSensitive":false,"CatalogName":"hive","Label":"_col0","Name":"_col0","Nullable":"UNKNOWN","Precision":10,"Scale":0,"SchemaName":"","TableName":"","Type":"integer"}]},"Rows":[{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]},{"Data":[{"VarCharValue":"1"}]}]},"UpdateCount":0,"UpdateType":""}
2017-08-11 08:11:56,472 - MainThread - botocore.hooks - DEBUG - Event needs-retry.athena.GetQueryResults: calling handler <botocore.retryhandler.RetryHandler object at 0x214e050>
2017-08-11 08:11:56,472 - MainThread - botocore.retryhandler - DEBUG - No retry needed.

**Comment by jguerrero77** [comment]
Thanks,

Yes, I understand. The problem is that the returned data is truncated with the data of the first call (1000 rows), so the data of the second POST call is missing. 
As a comparison, the same call on windows (with the old API) returns the first chunk (1000 rows) + a nextToken field, that I can use to call the function again.
At this point, in light of your comment, I am not sure if what I am getting from windows is the expected behavior, because it seems I am responsible for  requesting data until it is exhausted (no NextToken) .

**Comment by joguSD** [comment]
You would only be responsible for exhausting the data if you wanted to handle pagination manually. CLI support for athena pagination was added fairly recently, it may be possible that your versions on windows are ones that didn't have these operations auto-paginated and thus are acting as though `--no-paginate` is being passed. 

Furthermore, it's possible that the service returns less items than the desired `page-size` requested. For example, if page size is 1000 and max items is also 1000, the CLI may still have to make two api calls to get the 1000 max items. 

Are you sure the results from the second query are not included? How many results in total should there be?

**Comment by jguerrero77** [comment]
Hi again,
I understand the explanation about the windows, and that was actually the case (it was running a previous version of it). 
In regard to your question: Yes, I am sure the data for the second post is not included. I am only receiving exactly 1000 rows, and I should be receiving 1010 rows (I did the query on purpose to return exactly that, and from the Athena Console it does return 1010 rows).
Besides, I have upgraded the CLI both in windows and linux to 1.11.134, and now the results I get are both truncated to 1000.
